deeplake/util/transform.py

```diff
@@ -130,26 +104,84 @@ def store_data_slice_with_pbar(pg_callback, transform_input: Tuple) -> Dict:
     if isinstance(data_slice, deeplake.Dataset):
         data_slice = add_cache_to_dataset_slice(data_slice, tensors)
 
+    rel_tensors = [posixpath.relpath(tensor, group_index) for tensor in visible_tensors]
+
+    transform_dataset = TransformDataset(
+        rel_tensors,
+        all_chunk_engines,
+        group_index,
+        label_temp_tensors,
+        cache_size=cache_size,
+    )
+
     if extend_only:
-        extend_data_slice(
-            data_slice,
-            pipeline,
-            all_chunk_engines,
-            group_index,
-            pg_callback,
+        transform_fn = pipeline.functions[0]
+        extend_fn, args, kwargs = (
+            transform_fn.func,
+            transform_fn.args,
+            transform_fn.kwargs,
         )
-    else:
-        transform_data_slice_and_append(
-            data_slice,
-            pipeline,
-            visible_tensors,
-            label_temp_tensors,
-            actual_tensors,
-            all_chunk_engines,
-            group_index,
-            pg_callback,
-            skip_ok,
+        extend_fn(data_slice, transform_dataset, *args, **kwargs)
```

Comment:
this function is huge now, can we refactor?

Reply:
might be better to keep things together, and it doesnt look that big to me, tell me if this is a must

Reply:
A good rule of thumb in python is to keep most functions upto 30-40 lines of code. This is 100+, would strongly recommend refactor to increase code readability.